### ISSUE TITLE
Scalar operations

### ISSUE THREAD

**Issue by cquiroz** [erikerlandson/coulomb#211]
Should it be allowed to do scalar operations like multiply 2 meters by 4?
I guess addition and substraction are not possible with scalars but multiplication and division should be possible

**Comment by erikerlandson** [comment]
In general I have always been in OK with this idea, in fact scalars are equivalent to `Unitless`:

```scala
val q1 = (2.0).withUnit[Meter] * 2.0
// would be equivalent to
val q2 = (2.0).withUnit[Meter] * (2.0).withUnit[Unitless]
```

I had some problems trying to define left-multiplication by a scalar - the compiler did not like my attempts to overload operations on `Double` and friends, however right-multiply would work fine.

I could never quite decide if allowing scalar multiplication somehow undermined the type safety aspects, but if `coulomb` users want it I think it is pretty reasonable.

**Comment by erikerlandson** [comment]
closed by #307
